### PR TITLE
Protect parsing of data types from empty lines

### PR DESCRIPTION
**Pull request by lbetz**
If parsing empty rows in data type files the method first returns an nil class so strip doesn't work anymore.

### CODE REVIEW

**Comment by evgeni** [comment]
Thanks Lennart!

The rubocop failure is unrelated (fix in https://github.com/theforeman/kafo/pull/376).

Could you add a test for the empty line behaviour you're fixing?

File: lib/kafo/data_type_parser.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def initialize(manifest)
         line = line.force_encoding("UTF-8").strip
         next if line.start_with?('#')
 
-        line = line.split(' #').first.strip
+        line = line.split(' #').first.strip unless line.empty?
```
Comment (ekohl) [inline lib/kafo/data_type_parser.rb:18]: I had to check this, but what ends up happening is `''.split(' #')` returns `[]` so `[].first` is `nil`.

File: lib/kafo/data_type_parser.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def initialize(manifest)
         line = line.force_encoding("UTF-8").strip
         next if line.start_with?('#')
```
Comment (ekohl) [inline lib/kafo/data_type_parser.rb:16]: I'd prefer to implement this
```suggestion
        next if line.start_with?('#') || line.empty?
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by evgeni** [comment]
Applied suggestion and added a test.

**Review by bastelfreak — APPROVED** [review]

File: lib/kafo/data_type_parser.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ def initialize(manifest)
       type_line_without_newlines = +''
       manifest.each_line do |line|
         line = line.force_encoding("UTF-8").strip
-        next if line.start_with?('#')
+        next if line.start_with?('#') || line.empty?
```
Comment (evgeni) [inline lib/kafo/data_type_parser.rb:16]: This won't catch a line full of whitespace (which some editors might leave as is).
```suggestion
        next if line.start_with?('#')
        next if line.strip.empty?
```

WDYT?

Comment (evgeni) [reply lib/kafo/data_type_parser.rb:16]: oh wait, we already strip above, NVM

**Review by ekohl — APPROVED** [review]
Btw, long term I'd still prefer to make this depend on puppet-strings to extract the actual data types instead of attempting to maintain yet another parser. Short term this is a good fix.
